streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -201,106 +179,59 @@ private void addNewStandbyTasks(final Map<TaskId, Set<TopicPartition>> newStandb
         return tasks;
     }
 
-    /**
-     * Closes standby tasks that were not reassigned at the end of a rebalance.
-     *
-     * @return list of changelog topic partitions from revoked tasks
-     * @throws TaskMigratedException if the task producer got fenced (EOS only)
-     */
-    List<TopicPartition> closeRevokedStandbyTasks() {
-        final List<TopicPartition> revokedChangelogs = standby.closeRevokedStandbyTasks(revokedStandbyTasks);
-
-        // If the restore consumer is assigned any standby partitions they must be removed
-        removeChangelogsFromRestoreConsumer(revokedChangelogs, true);
-
-        return revokedChangelogs;
-    }
-
-    /**
-     * Closes suspended active tasks that were not reassigned at the end of a rebalance.
-     *
-     * @throws TaskMigratedException if the task producer got fenced (EOS only)
-     */
-    void closeRevokedSuspendedTasks() {
-        // changelogs should have already been removed during suspend
-        final RuntimeException exception = active.closeNotAssignedSuspendedTasks(revokedActiveTasks.keySet());
-
-        // At this point all revoked tasks should have been closed, we can just throw the exception
-        if (exception != null) {
-            throw exception;
-        }
-    }
-
     /**
      * Similar to shutdownTasksAndState, however does not close the task managers, in the hope that
      * soon the tasks will be assigned again.
-     * @return list of suspended tasks
      * @throws TaskMigratedException if the task producer got fenced (EOS only)
      */
-    Set<TaskId> suspendActiveTasksAndState(final Collection<TopicPartition> revokedPartitions)  {
-        final AtomicReference<RuntimeException> firstException = new AtomicReference<>(null);
-        final List<TopicPartition> revokedChangelogs = new ArrayList<>();
-
+    void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         final Set<TaskId> revokedTasks = partitionsToTaskSet(revokedPartitions);
-
-        firstException.compareAndSet(null, active.suspendOrCloseTasks(revokedTasks, revokedChangelogs));
-
-        changelogReader.remove(revokedChangelogs);
-        removeChangelogsFromRestoreConsumer(revokedChangelogs, false);
-
-        final Exception exception = firstException.get();
-        if (exception != null) {
-            throw new StreamsException(logPrefix + "failed to suspend stream tasks", exception);
+        for (final TaskId taskId : revokedTasks) {
```

Comment:
nit: maybe add a check that these tasks are actives only? Also do we guarantee that the task-manager would not host both active and standby of the same task-id?

Reply:
I was uneasy about that early on, but it should be impossible for a single instance of KaflaStreams, let alone one of its threads, to host more than one copy of a single task, active or otherwise.